lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -129,14 +129,42 @@ def generate_refresh_token
     end
 
     def generate_token
+      set_timestamps
+
       generator = Doorkeeper.configuration.access_token_generator.constantize
-      self.token = generator.generate(resource_owner_id: resource_owner_id,
-                                      scopes: scopes, application: application,
-                                      expires_in: expires_in)
+      self.token = generator.generate(
+        resource_owner_id: resource_owner_id,
+        scopes: scopes,
+        application: application,
+        expires_in: expires_in,
+        created_at: created_at
+      )
     rescue NoMethodError
       raise Errors::UnableToGenerateToken, "#{generator} does not respond to `.generate`."
     rescue NameError
       raise Errors::TokenGeneratorNotFound, "#{generator} not found"
     end
+
+    def set_timestamps
+      if created_at.nil?
+        self.created_at = current_time
```

Comment:
Normally yes but at the point when we call `generate_token` the timestamps have not yet been generated by rails and are `nil`.  For something like JWT where other systems are making decisions based on the `created_at` time burned into the token, it's important that doorkeeper's db agrees with the JWT token.